python/paddle/fluid/layers/nn.py

```diff
@@ -3815,3 +3816,39 @@ def roi_pool(input, rois, pooled_height=1, pooled_width=1, spatial_scale=1.0):
             "spatial_scale": spatial_scale
         })
     return pool_out
+
+
+def dice_loss(input, label, num_classes, reduce_dim=[1, 2, 3], epsilon=0.00001):
+    """
+    **Dice loss Layer**
+    Dice loss for comparing the similarity of two batch of data,
+    usually be used for binary image segmentation i.e. labels are binary.
+    The value between 0 to 1, 1 means totally match.
+
+    Args:
+        input (Variable): The predictions with shape [batch_size, ... , num_classes].
```

Comment:
The shape `[batch_size, ... , num_classes]` is not clear.  Is the rank large than 2 (>=2D)? The first dimension is batch size, the last dimension is class number?

Reply:
Fixed. Thx.